### DISCUSSION TITLE
Command History Not Shown In PowerShell 7

### DISCUSSION THREAD

**Discussion by rescenic** [JanDeDobbeleer/oh-my-posh#228] [Troubleshoot] [answered]
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] I looked for duplicate issues before submitting this one

### Description

PowerShell 5 can show the last command I typed. But PowerShell 7 can't do that.

### Environment

- Oh my Posh version: 3.51
- Theme: jandedobbeleer (Default Theme)
- Operating System: Windows 10 Insiders 2004 Build 20270.1
- Shell: pwsh 5 & 7
- Terminal: native windows

### Steps to Reproduce

1. Install using Scoop:
```
mkdir C:\tools
Invoke-Webrequest https://github.com/JanDeDobbeleer/oh-my-posh3/releases/latest/download/posh-windows-amd64.exe -OutFile C:\tools\oh-my-posh.exe
```

2. Edit both $PROFILE in respected powershell version folder:
PowerShell 5: C:\Users\User\Documents\WindowsPowerShell
PowerShell 7: C:\Users\User\Documents\PowerShell
```
[ScriptBlock]$Prompt = {
    $lastCommandSuccess = $?
    $realLASTEXITCODE = $global:LASTEXITCODE
    $errorCode = 0
    if ($lastCommandSuccess -eq $false) {
        #native app exit code
        if ($realLASTEXITCODE -ne 0) {
            $errorCode = $realLASTEXITCODE
        }
        else {
            $errorCode = 1
        }
    }
    $startInfo = New-Object System.Diagnostics.ProcessStartInfo
    $startInfo.FileName = "C:\tools\oh-my-posh.exe"
    $cleanPWD = $PWD.ProviderPath.TrimEnd("\")
    $startInfo.Arguments = "-config=""C:\tools\poshthemes\jandedobbeleer.json"" -error=$errorCode -pwd=""$cleanPWD"""
    $startInfo.Environment["TERM"] = "xterm-256color"
    $startInfo.CreateNoWindow = $true
    $startInfo.StandardOutputEncoding = [System.Text.Encoding]::UTF8
    $startInfo.RedirectStandardOutput = $true
    $startInfo.UseShellExecute = $false
    if ($PWD.Provider.Name -eq 'FileSystem') {
      $startInfo.WorkingDirectory = $PWD.ProviderPath
    }
    $process = New-Object System.Diagnostics.Process
    $process.StartInfo = $startInfo
    $process.Start() | Out-Null
    $standardOut = $process.StandardOutput.ReadToEnd()
    $process.WaitForExit()
    $standardOut
    $global:LASTEXITCODE = $realLASTEXITCODE
    #remove temp variables
    Remove-Variable realLASTEXITCODE -Confirm:$false
    Remove-Variable lastCommandSuccess -Confirm:$false
}
Set-Item -Path Function:prompt -Value $Prompt -Force
```

3. Reload profile / reenter powershell.

**Expected behavior:** It can show autosuggestions (command suggestion based on command history that I typed). oh-my-posh 2 can show that. But it broke, so I deleted everything about oh-my-posh2.

**Actual behavior:** PowerShell 7 can't show command suggestions.
![pwsh57](https://i.ibb.co/2Z3pbhH/09-12-2020-976-05-16-49.png)

**I tried to reinstall oh-my-posh3 based on: https://ohmyposh.dev/docs/installation#4-replace-your-existing-prompt, it didn't work at all**

**Comment by lnu** [comment]
Are you talking about command history or autocompletion of commands?  
I tried here with the latest version and when I type "$PS" and hit tab, the autocompletion takes place:
![temp](https://user-images.githubusercontent.com/1829553/101589153-29b73300-39e8-11eb-98e9-a5242dc62987.gif)

And for the profile, I use this:
```
import-module oh-my-posh
import-module posh-git
Import-Module psini
Set-PoshPrompt jandedobbeleer
```

it's easier than copy pasting the entier powershell block. It's useful if you want to mess around with it but since everything is in the powershell module, using it is easier.

**Reply by rescenic** [reply]
@lnu , Thanks for your reply. What I mean is this: 
![pwsh7](https://i.ibb.co/cr5jMzY/09-12-2020-985-19-15-43.png)
The suggestion appears after I type 1 word, then I can hit tab, it will autocomplete.

**Reply by lnu** [reply]
as you said, works fine with PSReadLine 2.1(`Set-PSReadLineOption -PredictionSource History`)

**Comment by JanDeDobbeleer** [comment, marked as answer]
@rescenic command completion has nothing to do with oh-my-posh (regardless of V2 or V3) as it's a feature enabled by [PSReadline](https://www.powershellgallery.com/packages/PSReadLine/2.1.0). Most likely that one's not set up correctly for your PS7.1 shell, there's more information to be found on that [here](https://devblogs.microsoft.com/powershell/announcing-psreadline-2-1-with-predictive-intellisense/).

Oh my Posh overwrites the `prompt` function to display a different styled prompt, and does not touch anything outside of that context (in line with the [UNIX philosophy](https://en.wikipedia.org/wiki/Unix_philosophy)).

**Reply by rescenic** [reply]
@JanDeDobbeleer Thank you for the hint. I tried PSReadline version 2.1.0, and it works now. But version 2.2.0-beta didn't work for PowerShell 7.1.

**Comment by omarabdelaz1z** [comment]
I have the same theme and I want to know how to remove "pwsh" and "no config"?



![image](https://user-images.githubusercontent.com/59033999/133160025-770f9ce1-dd42-4610-936f-2460733513f7.png)

This is the $profile
![image](https://user-images.githubusercontent.com/59033999/133160387-08f4835a-7f9b-4125-a653-162dcd224617.png)

**Reply by JanDeDobbeleer** [reply]
How did you install oh-my-posh?

**Reply by omarabdelaz1z** [reply]
I followed this guide [My Ultimate PowerShell prompt with Oh My Posh and the Windows Terminal](https://www.hanselman.com/blog/my-ultimate-powershell-prompt-with-oh-my-posh-and-the-windows-terminal) by Scott Hanselman.

**Reply by JanDeDobbeleer** [reply]
@omarabdelaz1z OK, so there are two things. Firstly, the `NO CONFIG` is because probably the `~\tonybaloney.omp.json` file is not there, so you might want to double-check as you see the default config, not the one that's loaded. Secondly, to adjust the segments (pwsh indication is the [shell](https://ohmyposh.dev/docs/shell) segment), you need to [tweak the config](https://ohmyposh.dev/docs/configure) (but in this case, first check if you have `~\tonybaloney.omp.json`).

**Reply by omarabdelaz1z** [reply]
I have added the __theme__.omp.json in the current-user directory and it works like a charm. 
Thank you man, Great theme btw.

![image](https://user-images.githubusercontent.com/59033999/133306924-19406fac-8eea-4f01-8a8d-2a3bd915bd4a.png)

**Reply by JanDeDobbeleer** [reply]
@omarabdelaz1z my pleasure, if you want to get rid of that space, have a look at [this](https://ohmyposh.dev/docs/faq#windows-terminal-unexpected-space-between-segmentstext) FAQ item.

**Reply by omarabdelaz1z** [reply]
You are king!
